File: apps/api-crm/Models/Promotion.cs
namespace Crm.Api.Models;

public class Promotion
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string PromotionType { get; set; } = string.Empty; // Discount, Voucher, FreeShipping, BuyOneGetOne, Cashback
    public decimal? DiscountValue { get; set; }
    public string? VoucherCode { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string Status { get; set; } = "Draft"; // Draft, Active, Cancelled, Accomplished
    public DateTime CreatedAt { get; set; }

    // Navigation properties
    public ICollection<CampaignPromotion> CampaignPromotions { get; set; } = [];
}

File: apps/web-crm/src/types/promotion.ts
export type PromotionType = "Discount" | "Voucher" | "FreeShipping" | "BuyOneGetOne" | "Cashback";
export type PromotionStatus = "Draft" | "Active" | "Cancelled" | "Accomplished";

export interface PromotionListItem {
  id: string;
  title: string;
  promotionType: PromotionType;
  status: PromotionStatus;
  discountValue?: number | null;
  endDate?: string | null;
  createdAt: string;
}

export interface Promotion extends PromotionListItem {
  description: string;
  voucherCode?: string | null;
  startDate?: string | null;
}

export interface CreatePromotionInput {
  title: string;
  description: string;
  promotionType: PromotionType;
  discountValue?: number;
  voucherCode?: string;
  startDate?: string;
  endDate?: string;
  status?: PromotionStatus;
}

export interface UpdatePromotionInput {
  title?: string;
  description?: string;
  promotionType?: PromotionType;
  discountValue?: number;
  voucherCode?: string;
  startDate?: string;
  endDate?: string;
  status?: PromotionStatus;
}
